mitodl/teachersportal
# Added API endpoint for CCXCon webhooks

#### What's this PR do?

Adds /api/v1/webhooks/ccxcon/ which is a write-only API for processing messages from CCXCon. These messages create, update, and delete courses and modules as they are stored in the product listing.
#### How should this be manually tested?

Update `portal/urls.py` to include the OSCAR app:

Add 

```
from oscar.app import application as oscar
```

And in the urls:

```
    url(r'^oscar/', include(oscar.urls)),
```

This will allow you to view oscar at `/oscar/` in the teacher's portal.

Then POST to the webhooks to create a course. The easiest way to do this is via jQuery. Go to `/api/v1/webhooks/ccxcon/` in the browser, and you should see a Django REST framework page saying `GET` is not allowed. Open the javascript console and run these lines:

```
$.ajax({url: '/api/v1/webhooks/ccxcon/', method: 'POST', contentType: 'application/json', data: JSON.stringify({type: 'Course', 'action': 'update', 'payload': {'title': 'Course title', 'external_pk': 'UUID goes here'}})}).then(function() {
  $.ajax({url: '/api/v1/webhooks/ccxcon/', method: 'POST', contentType: 'application/json', data: JSON.stringify({type: 'Module', 'action': 'update', 'payload': {'title': 'Module title', 'external_pk': 'ModuleUUID', 'course_external_pk': 'UUID goes here', 'subchapters': []}})})
});
```

Browse to the oscar product page and verify that you haven't hit on any weird errors, and that the module product is marked `Available`. (The course product will be marked unavailable.)

Also make sure that our UI is unaffected by these changes.
#### What are the relevant tickets?

Fixes #88
#### What gif best describes this PR or how it makes you feel?

http://i.imgur.com/YtebWvJ.gif
